### PR TITLE
Add linspace, logspace, geomspace, and arange variable constructors.

### LINKED ISSUE #1865: Numpy-like helpers? (linspace etc...)
It would be convenient to have additional array creation helpers which replicate the commonnly use `numpy` creators, namely `linspace`, `arange`, `logspace`, `geomspace`.

However, deciding on a syntax may not be as straightforward as other `numpy` functions such as `zeros`.
In `sc.zeros`, we have to specify the dims as well as the shape.

But `numpy` users are used to doing something like `sc.arange(20)`, or `sc.arange(3, 20)`. It might be annoying to always have to write `sc.arange(dim='x', start=3, stop=20)` or `sc.linspace(dim='y', start=3, stop=50, num=7)`?

Or maybe it's actually a good thing to have all the arg names because in `arange` and `linspace` I always forget if the 3rd number is the size of the steps or the number of steps.

### PR DESCRIPTION
**Pull request by nvaytet**
This adds the `numpy`-like variable constructors `linspace`, `logspace`, `geomspace`, and `arange`.

Notes:
- only keyword arguments are allowed, unlike numpy (because we also need a compulsory dimension label)
- only 1d variables are created: the `dim` argument expects a string, not a list (this should cover most use cases)

Fixes #1865

### CODE REVIEW

**Review by jl-wynen — COMMENTED** [review]
Minor point: I am not a fan of requiring keyword arguments here because they can easily more than double the length of a function call.

More importantly: All of these functions copy the numpy array into the Variable, see https://github.com/scipp/scipp/blob/efdee9e690ee4af22fb710398c0b4644c0a542a1/python/tests/test_ownership_variable.py#L211

File: docs/python-reference/api.rst [hunk 1]
```diff
@@ -28,6 +28,10 @@ Creation functions
    ones_like
    empty
    empty_like
+   arange
+   linspace
+   logspace
+   geomspace
```
Comment (SimonHeybrock) [inline docs/python-reference/api.rst:34]: Sort the list?

**Review by SimonHeybrock — COMMENTED** [review]
👍 
Can you add this to the release notes please?

**Comment by SimonHeybrock** [comment]
@jl-wynen wrote
> More importantly: All of these functions copy the numpy array into the Variable

I don't think this matters at this point? I am not aware of a practical use of, e.g., `linspace` with more than a couple million elements, i.e., performance should be irrelevant?

**Comment by SimonHeybrock** [comment]
> Minor point: I am not a fan of requiring keyword arguments here because they can easily more than double the length of a function call.

I am not sure where to draw the line. There is definitely big value in enforcing keyword arguments in most cases. Here, we can argue that maybe `dim`, `start`, and `stop` are "obvious"?
- Is it really always obvious?
- Can we force keywords for all the others, except for these 3?
- Is the little gained convenience worth it?

**Comment by nvaytet** [comment]
About the keyword arguments, I am aware of the potential annoyance, see #1865 
I don't know what is the best way to do it, and what to do with the `dim` argument?
Do you place it after all the other args `sc.arange(50, dim='x')`?

**Comment by SimonHeybrock** [comment]
> About the keyword arguments, I am aware of the potential annoyance, see #1865
> I don't know what is the best way to do it, and what to do with the `dim` argument?
> Do you place it after all the other args `sc.arange(50, dim='x')`?

I think `dim` should come first, and is acceptable as non-keyword (provided that we decide that `start` and `stop` are)?

File: python/src/scipp/_variable.py [hunk 2]
```diff
@@ -194,8 +194,7 @@ def array(*,
                     dtype=dtype)
 
 
-def linspace(*,
-             dim: str,
+def linspace(dim: str,
              start: Union[int, float],
              stop: Union[int, float],
              num: int,
```
Comment (SimonHeybrock) [inline python/src/scipp/_variable.py]: Now this seems to support non-keyword args for all args? I am not sure that is a good approach?

Comment (nvaytet) [reply python/src/scipp/_variable.py]: ah sorry, should I add the `*` before the scipp arguments?

Comment (SimonHeybrock) [reply python/src/scipp/_variable.py]: Not sure, see discussion I tried to start with https://github.com/scipp/scipp/pull/1871#issuecomment-827488778.

Comment (nvaytet) [reply python/src/scipp/_variable.py]: I think it's good like that, i think definitely people who use numpy are almost never using kwargs in these functions

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**